sql/rowexec/ddl_iters.go

```diff
@@ -494,27 +498,33 @@ func (i *modifyColumnIter) rewriteTable(ctx *sql.Context, rwt sql.RewritableTabl
 	}
 
 	var renames []sql.ColumnRename
-	if i.m.Column() != i.m.NewColumn().Name {
+	if oldColName != newCol.Name {
 		renames = []sql.ColumnRename{{
-			Before: i.m.Column(), After: i.m.NewColumn().Name,
+			Before: oldColName, After: newCol.Name,
 		}}
 	}
 
 	oldPkSchema := sql.SchemaToPrimaryKeySchema(rwt, rwt.Schema())
 	newPkSchema := sql.SchemaToPrimaryKeySchema(rwt, newSch, renames...)
 
 	rewriteRequired := false
-	if i.m.TargetSchema()[oldColIdx].Nullable && !i.m.NewColumn().Nullable {
+	if oldCol.Nullable && !newCol.Nullable {
+		rewriteRequired = true
+	}
+
+	oldEnum, isOldEnum := oldCol.Type.(sql.EnumType)
+	newEnum, isNewEnum := newCol.Type.(sql.EnumType)
+	if isOldEnum && isNewEnum {
```

Comment:
(minor) Seems like you may be able to optimize a little bit here and include something like `&& !oldEnum.Equals(newEnum)` – otherwise seems like cases like a column rename for an enum type that isn't changing would still trigger a rewrite? Not a huge deal of course, since altering a common isn't super common, but seems like this might be easy and safe to switch on.